src/mixins/ui/lasso-shapes/LatLonViewIntersectUtils.js

```diff
@@ -36,6 +37,289 @@ function canIntersect(numerator, denominator) {
+    // Early out:
+    // The line segment lies on a line of latitude or longitude
+    // Such lines do not need to be subdivided in web-mercator-projected
+    // space because they will be vertical/horizontal when projected (i.e.
+    // they will not exhibit any curvature)
+    // Return empty object.
+
+    // eslint-disable-next-line no-warning-comments
+    // TODO(croot): support other projections?
+
+    return rtn_obj
+  }
+
+  if (AABox2d.equals(intersect_aabox, line_aabox)) {
+    // Early out: the intersection bounding box equals the
+    // line segment's bounding box, meaning the entire line
+    // segment is in view. Add the start/end points as intersection
+    // points and mark the segment as needing subdividing.
+    rtn_obj.lonlat_pts.push(start_point_data.lonlat_point)
+    rtn_obj.screen_pts.push(start_point_data.screen_point)
+    rtn_obj.pts_t.push(0)
+    rtn_obj.lonlat_pts.push(end_point_data.lonlat_point)
+    rtn_obj.screen_pts.push(end_point_data.screen_point)
+    rtn_obj.pts_t.push(1)
+    rtn_obj.subdivide = true
+  } else {
+    // need to do intersection checks against the side of the intersection bounds. There will be
+    // exactly 2 intersections. The intersection check is a slightly simplified version of a
+    // bezier-based line segment intersection formula:
+    // https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection#Given_two_points_on_each_line_segment
+    // This formula can be slightly simplified due to the intersection bounds being axis-aligned.
+    // Ultimately you can eliminate some operations knowing that coordinate differences will be 0
+
+    if (
+      intersect_aabox[AABox2d.MINX] === intersect_aabox[AABox2d.MAXX] ||
+      intersect_aabox[AABox2d.MINY] === intersect_aabox[AABox2d.MAXY]
+    ) {
+      // early out: there's no intersection with the view bounds
+      // if the intersection aabox has no size in either dimension
+      // by the time we reach here. The case where the line is horizontal/vertical
+      // or the line is fully contained by the view should already be handled.
+      // return empty intersection object
+      return rtn_obj
+    }
+
+    const x3 = intersect_aabox[AABox2d.MINX]
+    const x4 = intersect_aabox[AABox2d.MAXX]
+    const y3 = intersect_aabox[AABox2d.MINY]
+    const y4 = intersect_aabox[AABox2d.MAXY]
+
+    const check_full_intersect = (
+      numerator_t_functor,
+      denominator_t_functor,
+      numerator_u_functor,
+      denominator_u_functor
+    ) => {
+      const denominator_t = denominator_t_functor()
+      if (denominator_t === 0) {
+        // colinear
+        // eslint-disable-next-line no-warning-comments
+        // TODO(croot): this needs filling out in the general case
+        // NOTE: this should never be hit in the LonLatPoly case because of
+        // the delta_x/delta_y === 0 check.
+
+        // eslint-disable-next-line no-console
+        console.assert(false, `Collinear intersection needs completing`)
```

Comment:
Are you from the midwest? I've been told that "<x> needs <y>[ing | ed]" is a midwest thing

Reply:
Oh copy paste from the midwest maybe...

Reply:
WAIT A MINUTE
`TODO(croot): this needs filling out in the general case`